### DISCUSSION TITLE
How to calculate GROD with respect to initial orientations

### DISCUSSION THREAD

**Discussion by Diego8901** [mtex-toolbox/mtex#1060] [Ask Anything] [answered]
Dear all,

I am conducting some in-situ experiments. I am wondering if there is any quick solution to calculate the Grain Reference Orientation Deviation (GROD) with respect to the initial orientation of the grain (before deformation), instead of the average orientation of the deformed grains. I have two columns of Grain IDs. The first column corresponds to the grains before deformation, and the second column corresponds to the same grains after deformation (due to twinning and grain subdivision, the same grains may have different IDs in the two scans).

Thanks!

Jie

**Comment by kilir** [comment, marked as answer]
Hi,
if I understand you correctly, also both lists of ids do not have the same length, i.e. you have more grains after deformation, right?
Is deformation low enough to still be able to identify pairs of pre- and post-deformation grains?

Cheers,
Rüdiger

**Reply by Diego8901** [reply]
Hi Rüdiger,

Thanks a lot for the reply. Sorry for not making my question clear enough. 

The deformation is low enough to identify MANY (but not all) the grain pairs (pre- and post deformation). As you said, I do have more grains after deformation. But this is not the major issue I am facing, because by manually selecting the grains (pre- and post deformation), I can make the two lists identical in length. The real problem is that when the function CalcGROD calculates GROD (grain REFERENCE ORIENTATION spread), it automatically assumes the reference orientation to be the average orientation of the deformed grain. However, I prefer using the grain orientation in the undeformed state as the reference orientation.

I can achieve this by doing a for loop: calculate the misorientation between each EBSD pixel in the deformed state and the average orientation of the grain in the undeformed state. But this makes the calculation rather slow. Thus, I would like to know if there are better solutions.

Cheers,
Jie

**Reply by kilir** [reply]
Hi Jie,
you can try something like the following:
```
mtexdata titanium
[grains, ebsd.grainId] = ebsd.calcGrains

% example: random ref orientation for each grain
% here you could use your list of old ref. orientations
meanOri = orientation.rand(length(grains))

% replicate to have for each ebsd point the appropriate ref. orientation
meanOri = meanOri(ebsd.grainId);
% compute misorientation
grod = inv(meanOri) .* ebsd.orientations;

plot(ebsd,grod.angle/degree)

```

Cheers,
Rüdiger

**Reply by Diego8901** [reply]
Appreciated, Rüdiger! It works!

Jie
